### PR TITLE
Account: Send a X-Request-ID header #5853

### PR DESCRIPTION
**Pull request by ckamm**

### CODE REVIEW

File: src/cmd/cmd.cpp [hunk 1]
```diff
@@ -311,7 +311,7 @@ int main(int argc, char **argv)
     qputenv("OPENSSL_CONF", opensslConf.toLocal8Bit());
 #endif
 
-    qsrand(QTime::currentTime().msec() * QCoreApplication::applicationPid());
+    qsrand(QDateTime::currentMSecsSinceEpoch());
```
Comment (guruz) [inline src/cmd/cmd.cpp]: Why did you remove this? I remember this was be added for smashbox doing parallel testing which had before this lead to duplicate transferids used in chunking

Comment (ckamm) [inline src/cmd/cmd.cpp]: Oh, makes sense to leave in the pid then. My complaint was that msec() just has values between 0-999.

**Comment by ckamm** [comment]
@ogoffart QUuid has curly braces, which can't appear. And not being limited to hex digits allows for shorter ids. Will update.

**Comment by guruz** [comment]
People on the internet seem to be using this header with smaller values separated by dashes.. (looks UUIDish..)

but no braces..

File: src/cmd/cmd.cpp [hunk 2]
```diff
@@ -311,7 +311,7 @@ int main(int argc, char **argv)
     qputenv("OPENSSL_CONF", opensslConf.toLocal8Bit());
 #endif
 
-    qsrand(QTime::currentTime().msec() * QCoreApplication::applicationPid());
+    qsrand(QDateTime::currentMSecsSinceEpoch() * QCoreApplication::applicationPid());
```
Comment (ogoffart) [inline src/cmd/cmd.cpp]: could this overflow? both calls return (signed) qint64  
  Epoch should be ~40 bits and i don't know how many bits is PID on windows for example.
could be worth casting to unsiged.

I would suggest   qHash(QDateTime::currentMSecsSinceEpoch()) ^ qhash(QCoreApplication::applicationPid())

**Comment by butonic** [comment]
Uuid is perfect as request id.

**Comment by butonic** [comment]
We could allow curly braces on the server side. Discuss with @PVince81 or @tomneedham while I am on vacation ;-)

**Comment by butonic** [comment]
Oh and yes, the request id must be uniqe for every request so it can be logged and traced through the whole server side stack.
